RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticHitToClusterAssociator.h

```diff
@@ -148,21 +150,39 @@ class RealisticHitToClusterAssociator
-                            hitToRealisticEnergyFraction_[hitId][clId] = assignedFraction;
-                            float assignedEnergy = assignedFraction *totalEnergy_[hitId];
-                            realisticSimClusters_[simClusterIndex].increaseEnergy(assignedEnergy);
-                            realisticSimClusters_[simClusterIndex].addHitAndFraction(hitId, assignedFraction);
-                            // if the hits energy belongs for more than exclusiveFraction to a cluster, also the cluster's
-                            // exclusive energy is increased. The exclusive energy will be needed to evaluate if
-                            // a realistic cluster will be invisible, i.e. absorbed by other clusters
-
-                            if( (useMCFractionsForExclEnergy and mcEnergyFraction_[hitId][clId] > exclusiveFraction) or
-                                    (!useMCFractionsForExclEnergy and assignedFraction > exclusiveFraction) ) {
-                              realisticSimClusters_[simClusterIndex].increaseExclusiveEnergy(assignedEnergy);
+                            if(assignedFraction > 1e-3)
+                            {
+                                hitToRealisticEnergyFraction_[hitId][clId] = assignedFraction;
+                                float assignedEnergy = assignedFraction *totalEnergy_[hitId];
+                                realisticSimClusters_[simClusterIndex].increaseEnergy(assignedEnergy);
+                                realisticSimClusters_[simClusterIndex].addHitAndFraction(hitId, assignedFraction);
+                                // if the hits energy belongs for more than exclusiveFraction to a cluster, also the cluster's
+                                // exclusive energy is increased. The exclusive energy will be needed to evaluate if
+                                // a realistic cluster will be invisible, i.e. absorbed by other clusters
+
+                                if( (useMCFractionsForExclEnergy and mcEnergyFraction_[hitId][clId] > exclusiveFraction) or
+                                        (!useMCFractionsForExclEnergy and assignedFraction > exclusiveFraction) )
+                                {
+                                    realisticSimClusters_[simClusterIndex].increaseExclusiveEnergy(assignedEnergy);
+                                }
+                            }
+                            else
+                            {
+                                removeAssociation.push_back(simClusterIndex);
                             }
                         }
                     }
                 }
+
+                while(!removeAssociation.empty())
+                {
+                    auto clusterToRemove = removeAssociation.back();
+                    removeAssociation.pop_back();
+                    auto it = std::find(hitToRealisticSimCluster_[hitId].begin(), hitToRealisticSimCluster_[hitId].end(), clusterToRemove);
```

Comment:
As a future development item, please find a faster way to do this than `std::find()`.